pilz_trajectory_generation/src/trajectory_generator.cpp

```diff
@@ -16,252 +16,237 @@
     }
-    if(req.goal_constraints.front().orientation_constraints.front().link_name.empty())
-    {
-      ROS_ERROR_STREAM("Link name of orientation constraint is empty. " );
-      error_code.val = moveit_msgs::MoveItErrorCodes::INVALID_GOAL_CONSTRAINTS;
-      return false;
-    }
-    if(req.goal_constraints.front().position_constraints.front().link_name !=
-       req.goal_constraints.front().orientation_constraints.front().link_name)
-    {
-      ROS_ERROR_STREAM("Target link name of position constraint: "
-                       << req.goal_constraints.front().position_constraints.front().link_name
-                       << " and orientation constraint: "
-                       <<  req.goal_constraints.front().orientation_constraints.front().link_name
-                       << " are different." );
-      error_code.val = moveit_msgs::MoveItErrorCodes::INVALID_GOAL_CONSTRAINTS;
-      return false;
-    }
-    // check the link name can be solved by ik
-    if(!robot_model_->getJointModelGroup(req.group_name)->
-       canSetStateFromIK(req.goal_constraints.front().position_constraints.front().link_name))
+
+    if ( !robot_model_->getJointModelGroup(group_name)->hasJointModel(curr_joint_name) )
     {
-      ROS_ERROR_STREAM("No IK solver is available for link: "
-                       << req.goal_constraints.front().position_constraints.front().link_name
-                       <<". ");
-      error_code.val = moveit_msgs::MoveItErrorCodes::NO_IK_SOLUTION;
-      return false;
+      std::ostringstream os;
+      os << "Joint \"" << curr_joint_name
+         << "\" does not belong to group \"" << group_name << "\"";
+      throw JointConstraintDoesNotBelongToGroup(os.str());
     }
-    // check if the needed pose is set
-    if(req.goal_constraints.front().position_constraints.front().constraint_region.primitive_poses.size() == 0)
+
+    if( !planner_limits_.getJointLimitContainer().verifyPositionLimit(curr_joint_name, joint_constraint.position) )
     {
-      ROS_ERROR_STREAM("No primitive poses available in posiion constraints. ");
-      error_code.val = moveit_msgs::MoveItErrorCodes::INVALID_GOAL_CONSTRAINTS;
-      return false;
+      std::ostringstream os;
+      os << "Joint \"" << curr_joint_name << "\" violates joint limits in goal constraints";
+      throw JointsOfGoalOutOfRange(os.str());
     }
   }
-
-  error_code.val = moveit_msgs::MoveItErrorCodes::SUCCESS;
-  return true;
-
 }
 
-bool TrajectoryGenerator::validateStartState(const planning_interface::MotionPlanRequest &req,
-                                             moveit_msgs::MoveItErrorCodes &error_code) const
+void TrajectoryGenerator::checkCartesianGoalConstraint(const moveit_msgs::Constraints& constraint,
+                                                       const std::string& group_name) const
 {
-  // check start state
-  if(req.start_state.joint_state.name.size() == 0)
+  assert(constraint.position_constraints.size() == 1);
```

Comment:
Please consider removing the asserts. Maybe something more robust and ready for production?

Reply:
I think the idea behind the asserts was misunderstood. There already exists a check [here](https://github.com/PilzDE/pilz_industrial_motion/blob/3cf90e044bf43642d8dd7ba374d147e33b988547/pilz_trajectory_generation/include/pilz_trajectory_generation/trajectory_generator.h#L252) and I am aware of this check. However, my intention here was to make sure the reader understands that at this place I assume a single element position and orientation constraint container and that the check should have happened beforehand.